core/reflection.jvm/src/kotlin/reflect/jvm/internal/KCallableImpl.kt

```diff
@@ -112,65 +112,80 @@ internal abstract class KCallableImpl<out R> : KCallable<R>, KTypeParameterOwner
         return if (isAnnotationConstructor) callAnnotationConstructor(args) else callDefaultMethod(args, null)
     }
 
+    private val _absentArguments = ReflectProperties.lazySoft {
+        val parameters = parameters
+        // arguments without masks and DefaultConstructorMarker or MethodHandle
+        val arguments = arrayOfNulls<Any?>(parameters.size + (if (isSuspend) 1 else 0))
+
+        // set absent values
+        parameters.forEach { parameter ->
+            if (parameter.isOptional && !parameter.type.isInlineClassType) {
+                // For inline class types, the javaType refers to the underlying type of the inline class,
+                // but we have to pass null in order to mark the argument as absent for InlineClassAwareCaller.
+                arguments[parameter.index] = defaultPrimitiveValue(parameter.type.javaType)
+            } else if (parameter.isVararg) {
+                arguments[parameter.index] = defaultEmptyArray(parameter.type)
+            }
+        }
+
+        arguments
+    }
+
+    private fun getAbsentArguments(): Array<Any?> = _absentArguments().clone()
+
     // See ArgumentGenerator#generate
     internal fun callDefaultMethod(args: Map<KParameter, Any?>, continuationArgument: Continuation<*>?): R {
         val parameters = parameters
-        val arguments = ArrayList<Any?>(parameters.size)
-        var mask = 0
-        val masks = ArrayList<Int>(1)
-        var index = 0
-        var anyOptional = false
 
-        for (parameter in parameters) {
-            if (index != 0 && index % Integer.SIZE == 0) {
-                masks.add(mask)
-                mask = 0
+        val arguments = getAbsentArguments().apply {
+            if (isSuspend) {
```

Comment:
I assume that the code below is also making the `isSuspend` determination, but is there any reason why it is not using the properties defined in the `KCallable` interface?
https://github.com/JetBrains/kotlin/blob/a366bd5e682133289647895aef2463f345a975cf/core/reflection.jvm/src/kotlin/reflect/jvm/internal/KCallableImpl.kt#L221

I think that need to fix the above code or adapt this implementation to the above code.

Reply:
There seems to be no reason. You're right, `KCallableImpl.extractContinuationArgument` can use `isSuspend` too.

Reply:
I was created PR to fix.
https://github.com/JetBrains/kotlin/pull/5038